libbeat/template/config.go

```diff
@@ -17,21 +17,61 @@
 
 package template
 
-import "github.com/elastic/beats/libbeat/common"
+import (
+	"github.com/elastic/beats/libbeat/common"
+)
+
+func Unpack(c *common.Config) (*TemplatesConfig, error) {
```

Comment:
We need to export this?

Why no `func (tc *TemplatesConfig) Unpack(c *common.Config) error` ? This would be used by go-ucfg.

Reply:
Basically because I didn't see a lot of value in manually creating an empty `TemplatesConfig` outside and then call the unpack function on it. But I see how this makes sense from a consistency point, so will change.